### PR TITLE
✨RTC & Analytics - Adding RTC and Analytics vars

### PR DESCRIPTION
**Pull request by ofirpaBrowsi**
This PR adds 5 RTC vars to the available vars that can be used in RTC vendor's request and AMP-Analytics 'elementId' var
### **amp-ad RTC vendor request vars**
- EL_TOP var is useful for determining where is the ad that asks for targeting
- SCROLL_TOP var is useful for understanding the user's position on the page
- PAGE_HEIGHT will help to know where is the ad in relation to the page
- BKG_STATE will be used to understand if ad requested when viewport is visible or hidden 
- PAGEVIEWID_64 var is relevant for correlation between events, for example - RTC request and amp-analytics.
### **amp-analytics events**
- ElementId var is addition to Visibility trigger so publisher could know which is the element that was visible.

### CODE REVIEW

**Comment by ofirpaBrowsi** [comment]
@googlebot I signed it!

**Comment by ofirpaBrowsi** [comment]
@googlebot I signed it!

**Comment by ofirpaBrowsi** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fampproject%2Famphtml%2Fpull%2F26266) for more info**.

@googlebot I signed it!

**Comment by ofirpaBrowsi** [comment]
@googlebot I signed it!

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -775,6 +777,12 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
           return this.element.getAttribute(name);
         }
       },
+      ELEMENT_POS: () => this.element.getPageLayoutBox().top,
```
Comment (lannka) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: @keithwrightbos this PR introduces more macros to RTC. I don't see an issue among those macros. But want to bring it to your attention.

File: extensions/amp-analytics/0.1/visibility-manager.js [hunk 2]
```diff
@@ -468,6 +468,7 @@ export class VisibilityManager {
       // Optionally, element-level state.
       let layoutBox;
       if (opt_element) {
+        state['elementId'] = opt_element.id;
```
Comment (lannka) [inline extensions/amp-analytics/0.1/visibility-manager.js]: @zhouyx any concerns?

Comment (zhouyx) [reply extensions/amp-analytics/0.1/visibility-manager.js]: I don't see any concern with this. 

There was proposal to include the element's tagName as well. Story team is adding story specific triggers. I'd propose that we use the same set of variable names in the future if we want to support elementId or similar. cc @gmajoulet @Enriqe

Comment (lannka) [reply extensions/amp-analytics/0.1/visibility-manager.js]: SGTM. what is used in story right now? shall we just follow the same naming pattern?

Comment (Enriqe) [reply extensions/amp-analytics/0.1/visibility-manager.js]: We currently don't have a custom variable for the elementId, only for tagName as seen here:

https://github.com/ampproject/amphtml/blob/0b9623cbc43521e28e395a58b9ab3613044cda14/extensions/amp-story/1.0/story-analytics.js#L25

File: extensions/amp-ad-network-doubleclick-impl/doubleclick-rtc.md [hunk 3]
```diff
@@ -28,9 +29,11 @@ Google Ad Manager's RTC implementation has made many macros available for RTC ur
 - **ATTR(data-override-width)** - data-override-width attribute of the amp-ad element
 - **ATTR(data-override-height)** - data-override-height attribute of the amp-ad element
 - **ATTR(data-json)** - data-json attribute of the amp-ad element
+- **TOP** - The offset of the element from document's top.
```
Comment (sebastianbenz) [inline extensions/amp-ad-network-doubleclick-impl/doubleclick-rtc.md]: nit: we should consistently end list items without `.` here

**Comment by lannka** [comment]
@ofirpaBrowsi could you address @sebastianbenz 's comments so we can merge this PR?

**Comment by ofirpaBrowsi** [comment]
@lannka fixed according to @sebastianbenz comments

**Pull request merged**